dask/array/slicing.py

```diff
@@ -428,13 +454,23 @@ def partition_by_size(sizes, seq):
     >>> partition_by_size([10, 20, 10], [1, 5, 9, 12, 29, 35])
     [[1, 5, 9], [2, 19], [5]]
     """
-    seq = np.array(seq)
-    right = np.cumsum(sizes)
-    locations = np.searchsorted(seq, right)
-    locations = [0] + locations.tolist()
-    left = [0] + right.tolist()
-    return [(seq[locations[i]:locations[i + 1]] - left[i]).tolist()
-            for i in range(len(locations) - 1)]
+    # seq = np.array(seq)
+    # right = np.cumsum(sizes)
+    # locations = np.searchsorted(seq, right)
+    # locations = [0] + locations.tolist()
+    # left = [0] + right.tolist()
+    # return [(seq[locations[i]:locations[i + 1]] - left[i]).tolist()
+    #         for i in range(len(locations) - 1)]
+
+    seq = np.asanyarray(seq)
+    left = np.empty(len(sizes) + 1, dtype=int)
+    left[0] = 0
+
+    right = np.cumsum(sizes, out=left[1:])
```

Comment:
The `out` parameter does add a little, but the objective was to avoid doing a concatenation to add the 0 as first element of `right` to create `left`, because that did take a lot of time.